bigchaindb_driver/offchain.py

```diff
@@ -0,0 +1,229 @@
+
+
+def prepare_transaction(*, operation='CREATE', owners_before=None,
+                        owners_after=None, asset=None, metadata=None,
+                        inputs=None):
+    """
+    Prepares a transaction payload, ready to be fulfilled. Depending on
+    the value of ``operation`` simply dispatches to either
+    :func:`~.prepare_create_transaction` or
+    :func:`~.prepare_transfer_transaction`.
+
+    Args:
+        operation (str): The operation to perform. Must be ``'CREATE'``
+            or ``'TRANSFER'``. Case insensitive. Defaults to ``'CREATE'``.
+        owners_before (:obj:`list` | :obj:`tuple` | :obj:`str`, optional):
+            One or more public keys representing the original owner(s)
+            of the asset being created. Only applies for ``'CREATE'``
+            operations. Defaults to ``None``.
+        owners_after (:obj:`list` | :obj:`tuple` | :obj:`str`, optional):
+            One or more public keys representing the new owner(s) of the
+            asset being created or transferred. Defaults to ``None``.
+        asset (:obj:`dict`, optional): The asset being created or
+            transferred. MUST be supplied for ``'TRANSFER'`` operations.
+            Defaults to ``None``.
+        metadata (:obj:`dict`, optional): Metadata associated with the
+            transaction. Defaults to ``None``.
+        inputs (:obj:`dict` | :obj:`list` | :obj:`tuple`, optional):
+            One or more inputs holding the condition(s) that this
+            transaction intents to fulfill. Each input is expected to
+            be a :obj:`dict`. Only applies to, and MUST be supplied for,
+            ``'TRANSFER'`` operations.
+
+    Returns:
+        dict: The prepared transaction.
+
+    Raises:
+        :class:`~.exceptions.BigchaindbException`: If ``operation`` is
+            not ``'CREATE'`` or ``'TRANSFER'``.
+
+    .. important::
+
+        * For ``operation='CREATE'``, ``owners_before`` MUST be set,
+          meanwhile ``owners_after``, ``asset``, amd ``metadata`` MAY be
+          set. The argument ``inputs`` is ignored.
+
+        * For ``operation='TRANSFER'``, ``owners_after``, ``asset``,
+          and ``inputs`` MUST be set, meanwhile  ``mwtadata`` MAY be set.
+          The argument ``owners_before`` is ignored.
+
+    """
+    operation = _normalize_operation(operation)
+    return _prepare_transaction(
+        operation,
+        owners_before=owners_before,
+        owners_after=owners_after,
+        asset=asset,
+        metadata=metadata,
+        inputs=inputs,
+    )
+
+
+def prepare_create_transaction(*,
+                               owners_before,
+                               owners_after=None,
+                               asset=None,
+                               metadata=None):
+    """
+    Prepares a ``"CREATE"`` transaction payload, ready to be
+    fulfilled.
+
+    Args:
+        owners_before (:obj:`list` | :obj:`tuple` | :obj:`str`): One
```

Comment:
Oh, this was changed. I think the change was made under https://github.com/bigchaindb/bigchaindb/commit/50b0b3cef217bc412dba5a062de1db2d1adcd514 but not sure.

@TimDaub and @r-marques may have more insights on this